### PR TITLE
Feat: support 'Enable Remote Apply' flag on environment settings

### LINKED ISSUE #761: Support Enable Remote Apply flag on environment settings
As part of remote apply feature, we let users to turn remote apply on or off for a remote backend environment, given few pre-conditions:

1. they have the EDIT_ALLOW_REMOTE_APPLY permission
2. remote backend is on for that environment
3. auto approval is on for that environment

In case one of those requirements are not met - turning on Remote Apply is not possible.

Relevant feature in the UI:
image

## DoD
- environment resource has a new field, `isRemoteApplyEnabled`.  its a boolean that defaults to false.
- [update environment endpoint](https://docs.env0.com/reference/environments-update) modified to support that field (only for existing resource)
- make sure that it is not enabled to set it as true if said pre conditions are not met

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #761 

### Solution

1. Added the schema and implemented the requirements.
2. Added additional acceptance tests.

### CODE REVIEW

File: env0/resource_environment_test.go [hunk 1]
```diff
@@ -185,6 +185,118 @@ func TestUnitEnvironmentResource(t *testing.T) {
...
+				LatestDeploymentLog: client.DeploymentLog{
+					BlueprintId: templateId,
+				},
+				IsRemoteApplyEnabled: true,
+			}
+
+			testCase := resource.TestCase{
+				Steps: []resource.TestStep{
+					{
+						Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:215]: regular creation.

File: env0/resource_environment_test.go [hunk 2]
```diff
@@ -185,6 +185,118 @@ func TestUnitEnvironmentResource(t *testing.T) {
...
+							resource.TestCheckResourceAttr(accessor, "approve_plan_automatically", "true"),
+							resource.TestCheckResourceAttr(accessor, "is_remote_apply_enabled", "true"),
+						),
+					},
+					{
+						Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+							"name":                       environment.Name,
+							"project_id":                 environment.ProjectId,
+							"template_id":                templateId,
+							"is_remote_backend":          !*environment.IsRemoteBackend,
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:258]: making this `false` to test the error use-case.

File: env0/resource_environment_test.go [hunk 3]
```diff
@@ -185,6 +185,118 @@ func TestUnitEnvironmentResource(t *testing.T) {
...
+							"force_destroy":              true,
+							"is_remote_apply_enabled":    true,
+						}),
+						ExpectError: regexp.MustCompile("cannot set is_remote_apply_enabled when approve_plan_automatically or is_remote_backend are disabled"),
+					},
+				},
+			}
+
+			runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+				gomock.InOrder(
```
Comment (TomerHeber) [inline env0/resource_environment_test.go:269]: alot of mocks... it's just how the provider works... /:

**Review by tomer-landesman — APPROVED** [review]
looks good to me 👍🏼

**Pull request merged**